build/lib/pgBackRestBuild/Config/Data.pm

```diff
@@ -0,0 +1,2234 @@
+        &CFGBLDOPT_RULE_REQUIRED => false,
+        &CFGBLDOPT_RULE_DEPEND =>
+        {
+            &CFGBLDOPT_RULE_DEPEND_OPTION => CFGBLDOPT_TEST,
+            &CFGBLDOPT_RULE_DEPEND_VALUE => true
+        },
+        &CFGBLDOPT_RULE_COMMAND =>
+        {
+            &CFGBLDCMD_ARCHIVE_PUSH => {},
+            &CFGBLDCMD_BACKUP => {},
+        }
+    },
+
+    # GENERAL Section
+    #-------------------------------------------------------------------------------------------------------------------------------
+    &CFGBLDOPT_ARCHIVE_TIMEOUT =>
+    {
+        &CFGBLDOPT_RULE_SECTION => CONFIG_SECTION_GLOBAL,
+        &CFGBLDOPT_RULE_TYPE => CFGBLDOPT_TYPE_FLOAT,
+        &CFGBLDOPT_RULE_DEFAULT => CFGBLDOPTDEF_DEFAULT_ARCHIVE_TIMEOUT,
+        &CFGBLDOPT_RULE_ALLOW_RANGE => [CFGBLDOPTDEF_DEFAULT_ARCHIVE_TIMEOUT_MIN, CFGBLDOPTDEF_DEFAULT_ARCHIVE_TIMEOUT_MAX],
+        &CFGBLDOPT_RULE_COMMAND =>
+        {
+            &CFGBLDCMD_ARCHIVE_PUSH => {},
+            &CFGBLDCMD_BACKUP => {},
+            &CFGBLDCMD_CHECK => {},
+        },
+    },
+
+    &CFGBLDOPT_BUFFER_SIZE =>
+    {
+        &CFGBLDOPT_RULE_SECTION => CONFIG_SECTION_GLOBAL,
+        &CFGBLDOPT_RULE_TYPE => CFGBLDOPT_TYPE_INTEGER,
+        &CFGBLDOPT_RULE_DEFAULT => CFGBLDOPTDEF_DEFAULT_BUFFER_SIZE,
+        &CFGBLDOPT_RULE_ALLOW_LIST =>
+        {
+            &CFGBLDOPTDEF_DEFAULT_BUFFER_SIZE_MIN         => true,
+            &CFGBLDOPTDEF_DEFAULT_BUFFER_SIZE_MIN * 2     => true,
+            &CFGBLDOPTDEF_DEFAULT_BUFFER_SIZE_MIN * 4     => true,
+            &CFGBLDOPTDEF_DEFAULT_BUFFER_SIZE_MIN * 8     => true,
+            &CFGBLDOPTDEF_DEFAULT_BUFFER_SIZE_MIN * 16    => true,
+            &CFGBLDOPTDEF_DEFAULT_BUFFER_SIZE_MIN * 32    => true,
+            &CFGBLDOPTDEF_DEFAULT_BUFFER_SIZE_MIN * 64    => true,
+            &CFGBLDOPTDEF_DEFAULT_BUFFER_SIZE_MIN * 128   => true,
+            &CFGBLDOPTDEF_DEFAULT_BUFFER_SIZE_MIN * 256   => true,
+            &CFGBLDOPTDEF_DEFAULT_BUFFER_SIZE_MIN * 512   => true,
+            &CFGBLDOPTDEF_DEFAULT_BUFFER_SIZE_MIN * 1024  => true,
+        },
+        &CFGBLDOPT_RULE_COMMAND =>
+        {
+            &CFGBLDCMD_ARCHIVE_GET => {},
+            &CFGBLDCMD_ARCHIVE_PUSH => {},
+            &CFGBLDCMD_BACKUP => {},
+            &CFGBLDCMD_CHECK => {},
+            &CFGBLDCMD_EXPIRE => {},
+            &CFGBLDCMD_INFO => {},
+            &CFGBLDCMD_LOCAL => {},
+            &CFGBLDCMD_REMOTE => {},
+            &CFGBLDCMD_RESTORE => {},
+            &CFGBLDCMD_STANZA_CREATE => {},
+            &CFGBLDCMD_STANZA_UPGRADE => {},
+        }
+    },
+
+    &CFGBLDOPT_DB_TIMEOUT =>
+    {
+        &CFGBLDOPT_RULE_SECTION => CONFIG_SECTION_GLOBAL,
+        &CFGBLDOPT_RULE_TYPE => CFGBLDOPT_TYPE_FLOAT,
+        &CFGBLDOPT_RULE_DEFAULT => CFGBLDOPTDEF_DEFAULT_DB_TIMEOUT,
+        &CFGBLDOPT_RULE_ALLOW_RANGE => [CFGBLDOPTDEF_DEFAULT_DB_TIMEOUT_MIN, CFGBLDOPTDEF_DEFAULT_DB_TIMEOUT_MAX],
+        &CFGBLDOPT_RULE_COMMAND =>
```

Comment:
CMD => false become complete obsolete some time back as far as I can see and honestly I never really understood your comment here (though it may have been valid at one time).  I change this to a hash because that's what it always gets translated to later on.

I've updated the comment to what I believe is correct.